Sources/StreamChat/Repositories/ConnectionRepository.swift

```diff
@@ -180,7 +175,9 @@ class ConnectionRepository {
         }
 
         let waiterToken = String.newUniqueId
-        connectionIdWaiters[waiterToken] = completion
```

Comment:
The change I did in the unit-test below captured the issue on this line. Unit-test would fail and says that many of the completion blocks were never called.